### ISSUE TITLE
Create API endpoint => src/pages/api/projects/index.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#475]
Create the API endpoint src/pages/api/projects/index.ts that will API route for getting all projects. Be sure to add the proper extension when creating the file. Here are the instructions:

## Overview

In this issue, we need to create a new Next.js API endpoint `index.ts` that will handle fetching all projects for a specific user. This endpoint will be used in the user's dashboard to display a list of all their projects.

## API Endpoint Details

- **Path**: `/api/projects`
- **Supported Method**: `GET`
- **Purpose**: Fetch all projects for the authenticated user.

## Authentication

The user must be authenticated to access this endpoint. The authentication is handled by NextAuth, and the user's session can be accessed using `import { getSession } from "next-auth/client";`. Make sure to check if the user is authenticated before fetching the projects. If the user is not authenticated, return a 401 Unauthorized status with a JSON object containing an error message.

## Input

The endpoint does not require any input parameters.

## Database Schema

The relevant database schema for this endpoint is the `Project` table. The schema for the `Project` table is defined in the `types.ts` file as follows:

```typescript
export const ProjectSchema = z.object({
  id: z.string().default("cuid()"),
  user_id: z.string(),
  name: z.string(),
  description: z.string().nullable(),
  language: z.string(),
  framework: z.string(),
  database: z.string(),
  hosting: z.string(),
  auth_framework: z.string(),
  css_styling: z.string(),
  slack_auth_token: z.string().nullable(),
  created_at: z.date().default(() => new Date()),
  updated_at: z.date().default(() => new Date()),
});

export type Project = z.infer<typeof ProjectSchema>;
```

## Database Interaction

Use the Prisma client to fetch all projects for the authenticated user. The Prisma client is already set up and can be imported using:

```typescript
import { prisma } from "~/server/db.ts";
```

The database query should fetch all projects where the `user_id` matches the authenticated user's ID. Use the `findMany` method from the Prisma client to fetch the projects.

## Data Validation

Before returning the fetched projects, validate the data using the Zod schema `ProjectSchema` from the `types.ts` file. Import the schema and use the `safeParse` method to validate the data.

## Output

The endpoint should return a JSON object containing an array of projects. Define a new TypeScript type for the output as follows:

```typescript
type ProjectsResponse = {
  projects: Project[];
};
```

The response should have a 200 OK status and include the array of projects in the JSON object.

## Error Cases

Handle the following error cases and return appropriate status codes and error messages:

1. If the user is not authenticated, return a 401 Unauthorized status with a JSON object containing an error message.
2. If there is an error fetching projects from the database or validating the data, return a 500 Internal Server Error status with a JSON object containing an error message.

## Summary

To summarize, the task is to create a new Next.js API endpoint `index.ts` at `/api/projects` that supports the `GET` method to fetch all projects for the authenticated user. The endpoint should use the Prisma client to interact with the database and validate the data using Zod before returning the response. Handle authentication and error cases as described above.
